pipeline-steps/artifact_build.groovy

```diff
@@ -1,71 +1,105 @@
-            credentialsId: "RPC_REPO_IP",
-            variable: "REPO_HOST"
-          ),
-          string(
-            credentialsId: "RPC_REPO_SSH_USERNAME_TEXT",
-            variable: "REPO_USER"
-          ),
-          file(
-            credentialsId: "RPC_REPO_SSH_USER_PRIVATE_KEY_FILE",
-            variable: "REPO_USER_KEY"
-          ),
-          file(
-            credentialsId: "RPC_REPO_SSH_HOST_PUBLIC_KEY_FILE",
-            variable: "REPO_HOST_PUBKEY"
-          )
-        ]){
+        get_rpc_repo_creds() {
+          aio_prepare.prepare()
           ansiColor('xterm') {
             dir("/opt/rpc-openstack/") {
               sh """#!/bin/bash
-              scripts/artifacts-building/python/build-python-artifacts.sh
+              scripts/artifacts-building/apt/build-apt-artifacts.sh
               """
             } // dir
           } // ansiColor
-        } // withCredentials
+        } // get_rpc_repo_creds
       } // withEnv
     } // stage
   ) // conditionalStage
 }
 
+def python(Map args) {
+  common.conditionalStage(
+    stage_name: "Build Python Artifacts",
+    stage: {
+      pubcloud.runonpubcloud {
+        try {
+          withEnv(args.environment_vars) {
+            get_rpc_repo_creds() {
+              aio_prepare.prepare()
+              ansiColor('xterm') {
+                dir("/opt/rpc-openstack/") {
+                  sh """#!/bin/bash
+                  scripts/artifacts-building/python/build-python-artifacts.sh
+                  """
+                } // dir
+              } // ansiColor
+            } // get_rpc_repo_creds
+          } // withEnv
+        } catch (e) {
+          print(e)
+          throw e
+        } finally {
+          common.archive_artifacts()
+        }
+      } // pubcloud slave
+    } // stage
+  ) // conditionalStage
+}
+
 def container(Map args) {
   common.conditionalStage(
     stage_name: "Build Container Artifacts",
     stage: {
-      withEnv(args.environment_vars) {
-        withCredentials([
-          string(
-            credentialsId: "RPC_REPO_IP",
-            variable: "REPO_HOST"
-          ),
-          string(
-            credentialsId: "RPC_REPO_SSH_USERNAME_TEXT",
-            variable: "REPO_USER"
-          ),
-          file(
-            credentialsId: "RPC_REPO_SSH_USER_PRIVATE_KEY_FILE",
-            variable: "REPO_USER_KEY"
-          ),
-          file(
-            credentialsId: "RPC_REPO_SSH_HOST_PUBLIC_KEY_FILE",
-            variable: "REPO_HOST_PUBKEY"
-          )
-        ]){
-          ansiColor('xterm') {
-            dir("/opt/rpc-openstack/") {
-              sh """#!/bin/bash
-              scripts/artifacts-building/containers/build-process.sh
-              """
-            } // dir
-          } // ansiColor
-        } // withCredentials
-      } // withEnv
+      pubcloud.runonpubcloud {
+        try {
+          withEnv(args.environment_vars) {
+            get_rpc_repo_creds() {
+              ansiColor('xterm') {
```

Comment:
This doesn't run an aio_prepare.prepare() like the other two functions do, meaning this stage cannot be run by itself.  I'm not sure if that's a problem at all (not all our stages work in isolation) but just mentioning it while I see it.